urlcdab7d8364_jrwalsh_CycTools_tgz-pJ8-edu_iastate_cyctools_MainJ8/src/edu/iastate/cyctools/view/MainCardPanel.java

```diff
@@ -0,0 +1,82 @@
+package edu.iastate.cyctools.view;
+
+import java.beans.PropertyChangeEvent;
+
+import edu.iastate.cyctools.DefaultController;
+import edu.iastate.cyctools.InternalStateModel.State;
+import edu.iastate.cyctools.externalSourceCode.AbstractViewPanel;
+
+import java.awt.CardLayout;
+import java.awt.Dimension;
+
+@SuppressWarnings("serial")
+public class MainCardPanel extends AbstractViewPanel {
+	DefaultController controller;
+	public final static String loginCard = "Login";
+	public final static String selectCard = "Select";
+	public final static String frameBrowseCard = "FrameBrowse";
+	public final static String exportCard = "Export";
+	public final static String searchCard = "Search";
+	public final static String structureExportCard = "Structure Export";
+	public final static String databaseCompareCard = "Database Compare";
+	public final static String loadCard = "Load";
+	
+
+	/**
+	 * Create the frame.
+	 * @param controller 
+	 */
+	public MainCardPanel(DefaultController controller) {
+		this.controller = controller;
+		initComponents();
+        localInitialization();
+    }
+
+    public void localInitialization() {
+    	//Add self as property change event listener of the controller
+    	controller.addView(this);
+    }
+    
+    private void initComponents() {
+    	this.setLayout(new CardLayout(0,0));
+	}
+    
+	@Override
+	public void modelPropertyChange(PropertyChangeEvent evt) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.